### PR TITLE
Increase reconcilers for Pod

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Increase the number of reconcilers for Pod in the default configuration and E2E tests.

Also doing the same for Workload object, which are expected to be a significant number of objects.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Increase the default number of reconcilers for Pod and Workload objects to 5, each.
```

### CODE REVIEW

**Comment by alculquicondor** [comment]
The error happened during cleanup and it doesn't seem related to this change. But I opened #1591 to investigate.

File: config/components/manager/controller_manager_config.yaml [hunk 1]
```diff
@@ -13,10 +13,11 @@ leaderElection:
 controller:
   groupKindConcurrency:
     Job.batch: 5
+    Pod.: 5
```
Comment (mimowo) [inline config/components/manager/controller_manager_config.yaml:16]: What is the default if not specified? Also, why only batch.Job and Pod frameworks are picked to have a custom value (5)?

**Review by mimowo — COMMENTED** [review]
The code changes LGTM, asked some questions to better understand the context.

File: config/components/manager/controller_manager_config.yaml (hunk 1 above)
Comment (alculquicondor) [reply config/components/manager/controller_manager_config.yaml:16]: The default is one.

We could add all integrations, but this is just a default configuration that should work for most users.

Comment (astefanutti) [reply config/components/manager/controller_manager_config.yaml:16]: Would that make sense it defaults according to the number of CPUs available / cgroups CPU quota?

Comment (alculquicondor) [reply config/components/manager/controller_manager_config.yaml:16]: This number is a default, users will have to pick up what makes sense for them.

We just have to pick some numbers that work at a reasonable scale (~50 nodes).
I picked the number based on the default number of workers in kcm.

**Pull request merged**